### ISSUE TITLE
Scripting API: Avoid throwing and catching CompilationErrorException in Script.Build

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#5949]

**Comment by ManishJayaswal** [comment]
@tmat what is the user impact of doing this workaround?

**Comment by tmat** [comment]
http://source.roslyn.io/#Microsoft.CodeAnalysis.Scripting/Script.cs,292

**Comment by tmat** [comment]
```
var diags = CSharpScript.Create("code with warnings").ContinueWith("1+1").Build();
```
Doesn't report warnings of the first code snippet in the resulting diagnostics.

**Comment by tmat** [comment]
I already have partial implementation. Just needs to be finished.

**Comment by tmat** [comment]
@ManishJayaswal Moved to 1.3
